app/src/main/java/com/github/h3lp3rs/h3lp/notification/NotificationService.kt

```diff
@@ -0,0 +1,121 @@
+                val descriptiontxt =
+                    ctx.resources.getString(R.string.notification_manager_description)
+                val importance = NotificationManager.IMPORTANCE_HIGH
+                val channel = NotificationChannel(CHANNEL_ID, title, importance).apply {
+                    description = descriptiontxt
+                }
+                val notificationManager: NotificationManager =
+                    ctx.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+                notificationManager.createNotificationChannel(channel)
+            }
+        }
+
+        /**
+         * Send simple notification without any action
+         * @param ctx the context of the app
+         * @param title Title on the notification
+         * @param description Description on the notification
+         */
+        fun sendSimpleNotification(ctx:Context,title : String, description : String){
+            val builder = buildBasicNotification(ctx,title,description)
+            sendNotification(builder.build(),ctx)
+        }
+
+        /**
+         * Send Notification with Intent triggered when clicked on
+         * @param ctx the context of the app
+         * @param title Title on the notification
+         * @param description Description on the notification
+         * @param intent the intent to trigger when the notification is clicked
+         */
+        fun sendIntentNotification(ctx:Context, title : String, description : String, intent : Intent){
+            val pendingIntent : PendingIntent = PendingIntent.getActivity(ctx,0,intent , 0)
+            val builder = buildBasicNotification(ctx,title,description)
+                .setContentIntent(pendingIntent)
+            sendNotification(builder.build(),ctx)
+        }
+
+        /**
+         * Send Notification opening an activity when clicked
+         * @param ctx the context of the app
+         * @param title Title on the notification
+         * @param description Description on the notification
+         * @param activity activity to open when the notification is clicked
+         */
+        fun <T> sendOpenActivityNotification(ctx:Context, title: String, description: String, activity: Class<T>){
+            val intent= Intent(ctx, activity).apply {
+                flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
+            }
+            sendIntentNotification(ctx,title,description,intent)
+        }
+
+        /**
+         * Send notification in with the given context
+         * Take care of assigning a unique id to the notification
+         * @param notification the notification to send
+         * @param ctx the app context from whem to throw notification
+         */
+        private fun sendNotification(notification : Notification, ctx: Context){
+            with(NotificationManagerCompat.from(ctx)) {
+                notify(notificationId, notification)
+            }
+            notificationId++
```

Comment:
You might want to make the increment atomic (just wrapping it into a synchronized block) since a same NotificationService could possibly be used concurrently